### PR TITLE
[onert] Handle a buffer shared by tensors during model loading

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
This enables `base_loader` to handle one buffer shared by multiple tensors.

For related comment : [1. here](https://github.sec.samsung.net/STAR/nnfw/issues/12049#issuecomment-1001199) and [2. here](https://github.sec.samsung.net/STAR/nnfw/issues/12049#issuecomment-1001574)

An example in circle or tflite model are as follows:

- model buffer
  - buffer 25 of **10 MB** weight
- subgraph 0:
  - tensor 1 points buffer 25
  - tensor 2 points buffer 25
- subgraph 1:
  - tensor 3 points buffer 25
  - tensor 4 points buffer 25
  
Current:
- `ir::Data` for tensor 1 is created and so does for tensor 2, 3, and 4,
  resulting in **40 MB** memory consumption.

With this commit:
- `ir::Data` for tensor 1 is created and tensor 2, 3, and 4 share
  the created `ir::Data`, resulting in **10 MB** memory consumption.

Signed-off-by: Hyun Sik Yoon <hyunsik.yoon@samsung.com>

### CODE REVIEW

**Review by hseok-oh — DISMISSED** [review]
LGTM

**Comment by hseok-oh** [comment]
@periannath PTAL

File: runtime/onert/frontend/base_loader/include/base_loader.h [hunk 1]
```diff
@@ -354,18 +358,43 @@ ir::OperandIndex BaseLoader<LoaderDomain>::loadOperand(const Tensor *tensor, ir:
...
       size_t mmap_size = offset_end - aligned_offset_start;
 
+      uint32_t buf_idx = tensor->buffer();
+      auto buffer_found = _buf_to_data.find(buf_idx);
+
       if (_use_mmaped_data)
       {
-        data_obj = std::make_unique<ir::MMapedData>(_fd, aligned_offset_start, mmap_size,
-                                                    unaligned_offset_start, data_size);
+        if (buffer_found != _buf_to_data.end())
```
Comment (periannath) [inline runtime/onert/frontend/base_loader/include/base_loader.h:366]: Optional) How about moving this if statement outside of `if (_use_mmaped_data)` to remove code duplication?

File: runtime/onert/frontend/base_loader/include/base_loader.h [hunk 2]
```diff
@@ -354,18 +358,43 @@ ir::OperandIndex BaseLoader<LoaderDomain>::loadOperand(const Tensor *tensor, ir:
...
+        {
+          // another tensor points this buffer and CachedData was already created
```
Comment (periannath) [inline runtime/onert/frontend/base_loader/include/base_loader.h:368]:
```suggestion
          // another tensor points this buffer and MMapedData was already created
```

Comment (hyunsik-yoon) [reply runtime/onert/frontend/base_loader/include/base_loader.h]: Fixed. :-) Thanks.

File: runtime/onert/frontend/base_loader/include/base_loader.h (hunk 1 above)
Comment (hyunsik-yoon) [reply runtime/onert/frontend/base_loader/include/base_loader.h]: Thanks. Fixed.

**Comment by hyunsik-yoon** [comment]
I also added @wateret as a reviewer since this change was discussed with him before.

**Review by periannath — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]

**Review by wateret — APPROVED** [review]
LGTM

**Pull request merged**